.github/workflows/build-linux.yml

```diff
@@ -48,6 +50,25 @@ jobs:
         ]
 
     steps:
+      - name: Prepare default inputs
```

Comment:
arbitrary line

I did a little digging and found [benc-uk/workflow-dispatch](https://github.com/benc-uk/workflow-dispatch) which can automatically trigger workflows using workflow_dispatch events (importantly for us with default input values so we don't have define default values multiple times). The docs say that you need a token with write access to the repo because GitHub prevents the default token from being able to fire the workflow_dispatch event, but that's no longer the case. `secrets.GITHUB_TOKEN` works just fine.

I'd also remove the `pull_request` event from the python package workflow. It doesn't really provide any value (you can pip install directly from GitHub), complicates the python version input handling, adds 2 more items to the already noisy PR events list, and tox already verifies that a python package can be built and installed on every PR (with all supported Python version and OS combinations).

If all of the build workflows are setup with workflow_dispatch events then we can reuse them from a release workflow. We can also extend this in the future by chaining jobs (test suites have to pass before builds run):

```
name: Release

on:
  release:
    types: [released]

jobs:
  release:
    runs-on: ubuntu-latest
    steps:
      - name: Trigger build and publish python package
        uses: benc-uk/workflow-dispatch@v1.1.0
        with:
          workflow: Build and publish python package
          token: ${{ secrets.GITHUB_TOKEN }}
      - name: Trigger Build for Linux
        uses: benc-uk/workflow-dispatch@v1.1.0
        with:
          workflow: Build for Linux
          token: ${{ secrets.GITHUB_TOKEN }}
      - name: Trigger Build for Mac
        uses: benc-uk/workflow-dispatch@v1.1.0
        with:
          workflow: Build for Mac
          token: ${{ secrets.GITHUB_TOKEN }}
      - name: Trigger Build for Windows
        uses: benc-uk/workflow-dispatch@v1.1.0
        with:
          workflow: Build for Windows
          token: ${{ secrets.GITHUB_TOKEN }}

```